status. native tarball is happy but site build fails, and it's related to something patching MetricsSystem.java

```
****************************************************************************
                          Maven Build and Install
****************************************************************************


$ /usr/bin/mvn -Dmaven.repo.local=/home/stevel/.m2/repository install -Pdist,src,yarn-ui -DskipTests -Dtar -Pnative -Drequire.snappy -Drequire.openssl -Drequire.fuse > /build/source/patchprocess/mvn_install.log 2>&1
$ cd /build/source
$ mv /build/source/hadoop-dist/target/hadoop-3.3.9-SNAPSHOT.tar.gz /build/source/target/artifacts/hadoop-3.3.9-SNAPSHOT.tar.gz
$ mv /build/source/hadoop-dist/target/hadoop-3.3.9-SNAPSHOT-src.tar.gz /build/source/target/artifacts/hadoop-3.3.9-SNAPSHOT-src.tar.gz


****************************************************************************
                                Maven Site
****************************************************************************


$ /usr/bin/mvn -Dmaven.repo.local=/home/stevel/.m2/repository install site site:stage -DskipTests -DskipShade -Pdist,src -Preleasedocs,docs > /build/source/patchprocess/mvn_site.log 2>&1

Failed!
```

from the output
```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.11.0:site (default-site) on project hadoop-common: failed to get report for org.apache.maven.plugins:maven-dependency-plugin: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project hadoop-common: Compilation failure
[ERROR] /build/source/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java:[67,25] method <T>register(java.lang.String,java.lang.String,T) is already defined in class org.apache.hadoop.metrics2.MetricsSystem
[ERROR]
```

git changes
```
!) git status
On branch build/HADOOP-18401-arm-64-branch-3.3
Your branch is up to date with 'github/build/HADOOP-18401-arm-64-branch-3.3'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	hadoop-common-project/hadoop-common/src/site/markdown/UnixShellAPI.md
	hadoop-common-project/hadoop-common/src/site/markdown/release/README.md
	hadoop-common-project/hadoop-common/src/site/markdown/release/index.md
	hadoop-common-project/hadoop-common/src/site/resources/configuration.xsl
	hadoop-common-project/hadoop-common/src/site/resources/core-default.xml
```

it looks like something patching the file, so on that second run it fails, as it is adding what is already there

```
git diff HEAD hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
index fef8c4b7e4b..7b21cc302d1 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
@@ -42,6 +42,18 @@ public abstract class MetricsSystem implements MetricsSystemMXBean {
   @InterfaceAudience.Private
   public abstract MetricsSystem init(String prefix);

+  /**
+   * Register a metrics source
+   * @param <T>   the actual type of the source object
+   * @param source object to register
+   * @param name  of the source. Must be unique or null (then extracted from
+   *              the annotations of the source object.)
+   * @param desc  the description of the source (or null. See above.)
+   * @return the source object
+   * @exception MetricsException
+   */
+  public abstract <T> T register(String name, String desc, T source);
+
   /**
    * Register a metrics source
    * @param <T>   the actual type of the source object
```

--- a/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch
+++ b/hadoop-common-project/hadoop-common/dev-support/jdiff-workaround.patch
@@ -1,5 +1,5 @@
 diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
-index a277abd6e13..1d131d5db6e 100644
+index fef8c4b7e4b..8d2b9339706 100644
 --- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
 +++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/MetricsSystem.java
 @@ -42,18 +42,6 @@
@@ -14,7 +14,7 @@ index a277abd6e13..1d131d5db6e 100644
 -   *              the annotations of the source object.)
 -   * @param desc  the description of the source (or null. See above.)
 -   * @return the source object
--   * @exception MetricsException
+-   * @exception MetricsException Metrics Exception.
 -   */
 -  public abstract <T> T register(String name, String desc, T source);
 -
@@ -38,7 +38,7 @@ index a277abd6e13..1d131d5db6e 100644
 +   *              the annotations of the source object.)
 +   * @param desc  the description of the source (or null. See above.)
 +   * @return the source object
-    * @exception MetricsException
+    * @exception MetricsException Metrics Exception.
     */
 -  public abstract <T extends MetricsSink>
 -  T register(String name, String desc, T sink);

